server/src/main/java/org/candlepin/resource/util/ConsumerInstalledProductEnricher.java

```diff
@@ -1,248 +0,0 @@
-import org.candlepin.policy.js.compliance.ComplianceStatus;
-
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
-/**
- * Responsible for enriching a {@link ConsumerInstalledProduct} with
- * product data.
- */
-public class ConsumerInstalledProductEnricher {
-
-    private static final String RED_STATUS = "red";
-    private static final String YELLOW_STATUS = "yellow";
-    private static final String GREEN_STATUS = "green";
-
-    private static final String[] GLOBAL_ATTRS = {"guest_limit"};
-
-    private ComplianceStatus status;
-    private Consumer consumer;
-    private ComplianceRules complianceRules;
-    private ProductCurator productCurator;
-
-    public ConsumerInstalledProductEnricher(Consumer consumer,
-        ComplianceStatus populatedComplianceStatus, ComplianceRules complianceRules,
-        ProductCurator productCurator) {
-        this.status = populatedComplianceStatus;
-        this.consumer = consumer;
-        this.complianceRules = complianceRules;
-        this.productCurator = productCurator;
-    }
-
-    /**
-     * Enrich a {@link ConsumerInstalledProduct} with data from the specified product.
-     *
-     * @param cip the installed product to enrich.
-     * @param prod the product to pull the data from.
-     */
-    public void enrich(ConsumerInstalledProduct cip, Product prod) {
-        if (cip.getVersion() == null) {
-            cip.setVersion(prod.getAttributeValue(Product.Attributes.VERSION));
-        }
-        if (cip.getArch() == null) {
-            cip.setArch(prod.getAttributeValue(Product.Attributes.ARCHITECTURE));
-        }
-        cip.setStatus(getStatus(prod.getId()));
-
-        DateRange range = getValidDateRange(prod);
-        cip.setStartDate(range != null ? range.getStartDate() : null);
-        cip.setEndDate(range != null ? range.getEndDate() : null);
-    }
-
-    /**
-     * Get the status of the product.
-     *
-     * @param prodId the id of the product to check.
-     * @return the status string
-     */
-    protected String getStatus(String prodId) {
-        String status = "";
-        if (this.status.getNonCompliantProducts().contains(prodId)) {
-            status = RED_STATUS;
-        }
-        else if (this.status.getPartiallyCompliantProducts()
-            .containsKey(prodId)) {
-            status = YELLOW_STATUS;
-        }
-        else if (this.status.getCompliantProducts().containsKey(
-            prodId)) {
-            status = GREEN_STATUS;
-        }
-        return status;
-    }
-
-    /**
-     * Get the {@link DateRange} that the specified product will be valid (inclusive).
-     *
-     * @param product the product
-     * @return the valid date range of the product.
-     */
-    protected DateRange getValidDateRange(Product product) {
```

Comment:
Yeah. Since the process so heavily relies on the work done in the rules, I just moved everything out to them to limit how many times we're serializing stuff.